subsys/net/lib/sockets/getaddrinfo.c

```diff
@@ -320,10 +321,17 @@ static inline int z_vrfy_z_zsock_getaddrinfo_internal(const char *host,
 #include <syscalls/z_zsock_getaddrinfo_internal_mrsh.c>
 #endif /* CONFIG_USERSPACE */
 
+#endif /* defined(CONFIG_DNS_RESOLVER) */
+
 int zsock_getaddrinfo(const char *host, const char *service,
 		      const struct zsock_addrinfo *hints,
 		      struct zsock_addrinfo **res)
 {
+	if (IS_ENABLED(CONFIG_NET_SOCKETS_OFFLOAD)) {
```

Comment:
This works fine for cc32xx, but I wonder if there are other platforms out there that support socket offload but not DNS offload, that would want to use the DNS resolver code below. Maybe it is nicer to check whether DNS functions are offloaded by doing something similar to the following:

```
if (socket_offload_dns_is_registered()) {  /* Check if an offloaded DNS implementation exists */
     return socket_offload_getaddrinfo(host, service, hints, res);
}
```
Then again, no one is currently asking for this, so I am ok with the way it is to have NET_SOCKETS_OFFLOAD imply DNS offload, and for the implementation to fail if DNS offload is not registered.